### ISSUE TITLE
Steps to new PR comment 100% release (per Q3 2023)

### ISSUE THREAD

**Issue by codecovdesign** [codecov/engineering-team#287]
_Epic outlines steps toward release of new comment, as per the Q3 review_

## Problem to solve: 
We get a lot of feedback that Codecov is noisy/not showing what’s important in both the PR comment and PR page in app. This epic is specifically relevant to the PR comment and the issue of all files being in the same table together, whether you touched them in your PR or not. We’re also looking to move the most important info to the top of the comment (clearly labeled patch and project coverage) as well as ameliorate the lack of indicators around removed code behavior. 

Updating the PR comment:

highlight patch and include project coverage up top. 

Include line to indicate when removed code behavior comes into play

Put indirect coverage changes minimized in its own section to highlight git diff files first and foremost

Improve routing to land on the specific file in the app that was clicked in the PR Comment

## Solution direction and planning

- layout prioritizing patch
     - block quote with data source and project change
     - small copy change removing by with :
- showing changed files separate from unexpected indirect changes
     - meets user expectations of "I change X files", previously from "why do I see Y amount?"
     - aligns to information architecture seen in upcoming pulls page
     - reduces cognitive load
- improves links routing user expects
     - specific file => lands on file expanded in pulls page
     - indirect changed files => lands on respective section in pulls page

**Applications team**
1. https://github.com/codecov/engineering-team/issues/295
      - follow up / in tandem w/ platform team: https://github.com/codecov/platform-team/issues/122
2. https://github.com/codecov/engineering-team/issues/286

**Other / pre-release**
1. https://github.com/codecov/engineering-team/issues/283
2. https://github.com/codecov/engineering-team/issues/284
5. https://github.com/codecov/engineering-team/issues/285
     - challenging to see in production

<details>
<summary>platform team related issues</summary>

- https://github.com/codecov/platform-team/issues/8
- https://github.com/codecov/platform-team/issues/9
- https://github.com/codecov/platform-team/issues/123
     - Feedback solicitation  applications-team#43
- https://github.com/codecov/platform-team/issues/122
     - blocked by: https://github.com/codecov/engineering-team/issues/295
</details>

**Comment by codecovdesign** [comment]
closing epic as all new changes have been made, captured and/or will be completed
